specification/managementgroups/resource-manager/Microsoft.Management/preview/2024-01-01-preview/management.json

```diff
@@ -0,0 +1,2383 @@
+          },
+          {
+            "$ref": "#/parameters/PatchGroupRequestParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ManagementGroup"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "PatchManagementGroup": {
+            "$ref": "./examples/PatchManagementGroup.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ManagementGroups"
+        ],
+        "operationId": "ManagementGroups_Delete",
+        "description": "Delete management group.\nIf a management group contains child resources, the request will fail.\n",
+        "parameters": [
+          {
+            "$ref": "#/parameters/GroupIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/CacheControlHeader"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted",
+            "schema": {
+              "$ref": "#/definitions/AzureAsyncOperationResults"
+            },
+            "headers": {
+              "Location": {
+                "description": "URL for determining when an operation has completed. Send a GET request to the URL in Location header.\nThe URI should return a 202 until the operation reaches a terminal state and 200 once it reaches a terminal state.\n\nFor more info: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#202-accepted-and-location-headers",
+                "type": "string"
+              },
+              "Azure-AsyncOperation": {
+                "description": "URL for checking the ongoing status of the operation.\nTo get the status of the asynchronous operation, send a GET request to the URL in Azure-AsyncOperation header value.\n\nFor more info: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations",
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "NoContent"
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DeleteManagementGroup": {
+            "$ref": "./examples/DeleteManagementGroup.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        }
+      }
+    },
+    "/providers/Microsoft.Management/managementGroups/{groupId}/descendants": {
```

Comment:
>/providers/Microsoft.Management/managementGroups/{groupId}/descendants [](http://example.com/codeflow?start=5&length=70)

To ARM, this models a collection GET of a resource type so it should return a collection of instances of managementGroups/descendants"(and probably also have GET, PUT and DELETE on /providers/Microsoft.Management/managementGroups/{groupId}/descendants/{descendantName}). But it's not actually a resource type, since it returns a list of something that's not a resource type (managementGroup or subscription Ids). I think this should be modeled either as an actual resource type or as a POST action 'getDescendents' on the management group instance.